### PR TITLE
Added AWS Lambda stream type to `aws-lambda`.

### PR DESCRIPTION
**Pull request by lancejpollard**
Getting the ball rolling on this, need some help to complete, not sure exactly if I got it right (plus then need to test). Please hint at the types of tests I should add. And I wasn't 100% sure I nailed the types correctly, the code is kind of messy haha. But I am 99% sure.

The types I added come from https://github.com/aws/aws-lambda-nodejs-runtime-interface-client/blob/7374a4e338fc3b070811d55ee337740f3f6cb382/src/ResponseStream.js#L37-L64

- Hints at the API: https://betterdev.blog/lambda-response-streaming-flush-content/

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
  - I am not sure about this, did I do things right? How do I import `stream` properly?
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

File: types/aws-lambda/handler.d.ts [hunk 1]
```diff
@@ -170,3 +176,36 @@ export interface ClientContextEnv {
...
+  HttpResponseStream: {
+    from: (
+      stream: ResponseStream,
+      prelude: ResponseStreamPrelude,
+    ) => ResponseStream
+  }
+}
+
+export type ResponseStream = stream.Writable &
+  stream.EventEmitter & {
```
Comment (codershiba) [inline types/aws-lambda/handler.d.ts:200]:
```suggestion
  typeof stream.EventEmitter & {
```

File: types/aws-lambda/handler.d.ts [hunk 2]
```diff
@@ -170,3 +176,36 @@ export interface ClientContextEnv {
...
  */
 export type Callback<TResult = any> = (error?: Error | string | null, result?: TResult) => void;
+
+export type StreamHandler = (
+  event: APIGatewayProxyEvent,
+  stream: ResponseStream,
+  context: Context,
+) => Promise<void>
+
+declare var awslambda: {
```
Comment (codershiba) [inline types/aws-lambda/handler.d.ts:186]:
```suggestion
export var awslambda: {
```

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/aws-lambda/handler.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 9f4b9deaf53b83a3f7b5422c4ce95502505af893
</p>

File: types/aws-lambda/handler.d.ts [hunk 3]
```diff
@@ -170,3 +176,36 @@ export interface ClientContextEnv {
  *   Pass `null` or `undefined` for the `error` parameter to use this parameter.
  */
 export type Callback<TResult = any> = (error?: Error | string | null, result?: TResult) => void;
+
+export type StreamHandler = (
+  event: APIGatewayProxyEvent,
```
Comment (simonbuchan) [inline types/aws-lambda/handler.d.ts:181]: I don't think the event has to be from API Gateway. In fact, API Gateway doesn't support streaming the response to reduce TTFB, IIRC.

This should be generic in the event like Handler, and ideally every trigger would have a streaming alias too, but that's probably a bit too much to ask for.

**Review by simonbuchan — CHANGES_REQUESTED** [review]
small nit to avoid compatibility issues in the future

sorry I didn't see this one earlier!

**Pull request closed without merging**